lib/hammer_cli_foreman/commands.rb

```diff
@@ -522,6 +527,22 @@ def self.desc(desc=nil)
       description.strip.empty? ? _("Associate a resource") : description
     end
 
+    def self.failure_message(msg = nil)
+      if msg.nil?
+        super(default_message "Could not associate the %{resource_name}")
+      else
+        super msg
+      end
+    end
```

Comment:
We use the massage methods in DSL manner. When no argument is passed we suppose it's called as a getter. Please see implementation of `super` or `self.desc` above. Therefore the overriding method should have IMHO been:
```ruby
def self.failure_message(msg = nil)
  super || default_message(_("Could not associate the %{resource_name}"))
end
```
The current implementation always hides the original messages that were set in the command.